includes/amp-helper-functions.php

```diff
@@ -438,3 +454,18 @@ function amp_prepare_comment_post() {
 	$origin = esc_url_raw( wp_unslash( $_GET['__amp_source_origin'] ) ); // WPCS: CSRF ok.
 	header( 'AMP-Access-Control-Allow-Source-Origin: ' . $origin, true );
 }
+
+/**
+ * Handle a general, non comment AMP XHR post.
+ *
+ * @since 0.7.0
+ * @param string $location The location to redirect to.
+ */
+function amp_handle_general_post( $location ) {
+
+	$url = site_url( $location );
```

Comment:
@westonruter Sorry I always get those two mixed up. was supposed to be `home_url()` since the `$location` is a relative link. But this may or maynot be, so should check for host first.
Good catch :)